swaybar/ipc.c

```diff
@@ -8,7 +8,17 @@
 #include "bar/ipc.h"
 
 void ipc_send_workspace_command(const char *workspace_name) {
-	sway_log(L_DEBUG, "Clicked on window %s", workspace_name);
+	sway_log(L_DEBUG, "Clicked on workspace %s", workspace_name);
+	size_t size = strlen("workspace ") + strlen(workspace_name) + 1;
+
+	char *command = malloc(size);
+	if (!sway_assert(command != NULL, "Failed to allocate memory for workspace command")) {
+		return;
+	}
+	snprintf(command, size, "workspace %s", workspace_name);
+
+	uint32_t reply_size;
+	ipc_single_command(swaybar.ipc_socketfd, IPC_COMMAND, command, &reply_size);
```

Comment:
And another bug, the last argument to ipc_single_command is both input and output, even though the comment says:

`
/**             
 * Issues a single IPC command and returns the buffer. len will be updated with
 * the length of the buffer returned from sway.
 */ 
`